209 new build

anton-seaice opened the pull request:
Draft of new versions for #209

anton-seaice commented:
@chrisb13 - we can add the MOM version to this branch too. It doesn't build yet. There is some more to chase down in CESM_share still

(It gives this mysterious error:
```
 >> 547    /g/data/tm70/as2285/spack0.22/environments/x/access-om3-nuopc/share/CESM_share/src/shr_reprosum_mod.F90:453: undefined reference to `t_barrierf_'
     548    share/libOM3_share.a(shr_reprosum_mod.F90.o): In function `shr_reprosum_mod_mp_shr_reprosum_calc_':
  >> 549    /g/data/tm70/as2285/spack0.22/environments/x/access-om3-nuopc/share/CESM_share/src/shr_reprosum_mod.F90:453: undefined reference to `t_barrierf_'
     550    share/libOM3_share.a(shr_reprosum_mod.F90.o): In function `shr_reprosum_mod_mp_shr_reprosum_calc_':
  >> 551    /g/data/tm70/as2285/spack0.22/environments/x/access-om3-nuopc/share/CESM_share/src/shr_reprosum_mod.F90:453: undefined reference to `t_barrierf_'
```
)

anton-seaice commented on CDEPS/patches/atm_comp_nuopc.F90.patch:
```diff
@@ -1,27 +1,27 @@
           trim(datamode) == 'CORE2_IAF'    .or. &
           trim(datamode) == 'CORE_IAF_JRA' .or. &
-+         trim(datamode) == 'JRA55do'.or. &
++         trim(datamode) == 'JRA55do     ' .or. &
```
```suggestion
+         trim(datamode) == 'JRA55do'      .or. &
```

anton-seaice commented:
If https://github.com/ESCOMP/CESM_share/pull/58 gets merged we can remove the #ifdef TIMING patch again.

anton-seaice commented:
If https://github.com/ESCOMP/WW3/pull/33 gets merged we can remove the ifdef W3_IC4 patch too

anton-seaice commented:
I also asked about the wrong variable name in CDEPS [here](https://github.com/ESCOMP/CDEPS/pull/316/files#)

I moved a newer CDEPS because it's sets a default value for a namelist parameter we don't set, (https://github.com/ESCOMP/CDEPS/pull/315)

anton-seaice commented on MOM6/CMakeLists.txt:
```diff
@@ -364,17 +364,34 @@ target_sources(OM3_mom6 PRIVATE
+  MOM6/src/equation_of_state/MOM_EOS_base_type.F90
+  MOM6/src/parameterizations/lateral/MOM_self_attr_load.F90
+
   extra_sources/mom_cap_gtracer_flux.F90
```
@chrisb13 - I think extra_sources should go in our mom6 branch also ?

chrisb13 replied:
As discussed, now added [here](https://github.com/ACCESS-NRI/MOM6/commit/9af8a789b895055ee1fd02ee667d07cfc5cacf5d). 

Note that I didn't both updating my build branch [here](https://github.com/COSIMA/access-om3/blob/cbull_mom6_updates/MOM6/CMakeLists.txt) as I think you are building elsewhere?

anton-seaice replied:
You can update this branch - we can tidy up the commits before merging if needed and then try a redeploy [here](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/33)

chrisb13 replied:
Okay, now [running](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/33#issuecomment-2571812183).

anton-seaice commented:
@aekiss @chrisb13 

This is basically ready for review. This change implements new component versions for the model components as follows:

- MOM6, using an up to date MOM-ocean version + the generic tracers changes. Access-nri version [2025.01.0](https://github.com/ACCESS-NRI/MOM6/releases/tag/2025.01.0)
- CICE, using CICE6.6.0, plus local changes for MOM supergrid implementation and to fix cice history filenames per https://github.com/COSIMA/access-om3/issues/190#issuecomment-2261753828. Once https://github.com/ACCESS-NRI/CICE/pull/9 is resolved I will do a release tag.
- WW3, [dev/unified_0.0.12 from CESM](https://github.com/ESCOMP/WW3/commits/1c11fc175793cb52a8bb4ebac70623a813fe00cd/) plus a local patch to fix a minor implementation issue with the "IC4_NUMERICS" fix
- CDEPS 1.0.59 from CESM
- CMEPS 1.0.25 from CESM
- CESM_share 1.1.5 from CESM

See individual commits for minor implementation details.



See pre-release deployment: 

https://github.com/ACCESS-NRI/ACCESS-OM3/pull/33

and 

Draft config update:

https://github.com/ACCESS-NRI/access-om3-configs/pull/154#issuecomment-2576295508

aekiss commented on CDEPS/extra_sources/datm_datamode_jra55do_mod.F90:
```diff
@@ -283,39 +281,4 @@ subroutine datm_datamode_jra55do_advance(exportstate, target_ymd, target_tod, mo
   end subroutine datm_datamode_jra55do_advance
 
-  !===============================================================================
-  subroutine datm_datamode_jra55do_restart_write(case_name, inst_suffix, ymd, tod, &
```
is it ok this these 2 subroutines are deleted?

anton-seaice replied:
Yeah its related to [this upstream change](https://github.com/ESCOMP/CDEPS/commit/863def0c8a6295dbbb4b44b1055845d980cd5978)

The restart write and read subroutines were generalised so that is not a specific subroutine for each data product, just one in `dshr_mod`

aekiss reviewed (approved):
LGTM on a quick look, but I'm too far removed from the nitty-gritty to pick much up. Maybe @dougiesquire could have a look if he's back?

chrisb13 reviewed (commented):
[LGTM too](https://github.com/ACCESS-NRI/access-om3-configs/pull/154#issuecomment-2591205258), although I'm most familiar with the MOM changes. Small things...

To be clear, I've understood this PR to involve these 3 sets of changes (22 commits), plus model component updates, which would all be accepted/merged at once (correct?):
- https://github.com/cosima/ACCESS-OM3/compare/main...209-new-build
- https://github.com/ACCESS-NRI/ACCESS-OM3/compare/main...209-new-cosima-build-2
- https://github.com/ACCESS-NRI/access-om3-configs/compare/dev-1deg_jra55do_ryf...209-dev-1deg_jra55do_ryf

On a practical point, once everything is merged (assuming you'll do the default GH 'create merge commit'), it's not clear to me [how this reference to](https://github.com/ACCESS-NRI/ACCESS-OM3/compare/main...209-new-cosima-build-2#diff-e8582e74fa156f4e5729a850e52b24f2fde2d815c2c9c360f88c4cf90db851abR55) `209-new-build` will work? @micaeljtoliveira tells me the branch name will remain where it is now but the CI may choose to work off the merge commit, which in some cases could be problematic. I see how it's convenient to reference a branch when testing an evolving build but I think now that it's at a release it is safest for this to be changed to a tagged commit. I also think a tagged commit should be used for CICE (perhaps a preliminary one whilst you're awaiting resolution on [this](https://github.com/ACCESS-NRI/CICE/pull/9)).

What is the reason for [this failing](https://github.com/ACCESS-NRI/ACCESS-OM3/pull/33#issuecomment-2597395743)? I note that the [repro check](https://github.com/ACCESS-NRI/access-om3-configs/pull/154/files#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034R53) actually uses `pr33-12` (not the failing `pr33-14`). (Although, the [symmetric option](https://github.com/ACCESS-NRI/access-om3-configs/pull/155#issuecomment-2594044087) has been tested elsewhere.) In general, can we please write a brief status when running CI commands such that there's a record of what's changed/why it's being run.

Since @dougiesquire is back on Monday, perhaps we should ask him to have a quick look too?

chrisb13 commented:
@anton-seaice. Thanks for clarifying, sounds good.

> Yes 209-new-build gets deleted. Ill will do a "release" in this repository, creating an 0.4.0 tag. The 0.4.0 tag then gets used for the deployment repo.

Ok, let's have a chat about this next week. Would be good to make sure we're on the same page -- to my mind, the safest option is to make the tag _before_ the merge (is that what you mean?).

anton-seaice commented:
At the team meeting today - we decided to use patches for MOM6 in this release instead of a commit from a branch.

I've changed MOM6 to pull the latest commit from MOM-ocean, and reverted the patches. The build fails now and I assume that @dougiesquire and or @chrisb13 will fix the patch files in the MOM6/patches folder

anton-seaice commented:
It offers be a delete option for the `2025.01.0` Pre-release [here](https://github.com/ACCESS-NRI/MOM6/releases) of MOM6. Shall I do that?

dougiesquire commented:
> It offers me a delete option for the 2025.01.0 Pre-release [here](https://github.com/ACCESS-NRI/MOM6/releases) of MOM6. Shall I do that?

Probably yes, but maybe worth holding off until we have a working alternative

dougiesquire commented:
Okay, okay, I'll get this working locally first 😑...

dougiesquire commented:
I've redone the patch files for the recent MOM update (now up to date with current `mom-ocean/MOM6:main`). The MOM6 test suite now passes with these patches applied (bearing in mind that `MOM_couplertype_infra.F90.patch` needs to be applied to both `config_src/infra/FMS1` and `config_src/infra/FMS2` for the tests) - see [this branch](https://github.com/ACCESS-NRI/MOM6/tree/PRcandidate-202502-gtracer-nuopc) of our MOM6 fork with the patches enabling generic tracers applied.

xref https://github.com/ACCESS-NRI/MOM6/issues/5

dougiesquire reviewed (commented):
Thanks @anton-seaice et al - looks great! One question, one suggestion.

anton-seaice commented on WW3/patches/w3srcemd.F90.patch:
```diff
@@ -0,0 +1,22 @@
+ #ifdef W3_IC3
+       CALL W3SIC3 ( SPEC,DEPTH, CG1,  WN1, IX, IY, VSIC, VDIC )
+ #endif
+-#ifndef W3_IC4_NUMERICS
++#if defined(W3_IC4) && !defined(W3_IC4_NUMERICS)
```
There was a bug identified a while ago in the Numerics of the implementation of the IC4 wave-ice attenuation scheme in WW3. In the ESCOMP fork, a change was made to add the W3_IC4_NUMERICS CPP flag which when turns on builds WW3 with the Numerics fixed for the IC4 scheme.

However, we don't have W3_IC4 set at this point (our default is a different wave-ice attenuation scheme) and it doesn't build without this change. When the IC4_NUMERICS was added, a couple of places were changed from `#ifdef W3_IC4` to `#ifndef W3_IC4_NUMERICS`, e.g. 

https://github.com/ESCOMP/WW3/commit/acfcfb7a02426869211d8cca39ea1928dc810272#diff-20540f78ddbaa883e664290c2bde4030b0eeb813fc9dc1aa50a5aa928789192dL2116-R2152

However this is incorrect. W3_IC4_NUMERICS should only be used when W3_IC4 is set, so it needs the change above

I raised https://github.com/ESCOMP/WW3/pull/33 - it looks like in the offical WW3 they might fix it a different way

dougiesquire replied:
Thanks!

dougiesquire replied:
Sorry, just realised I missing [this comment](https://github.com/COSIMA/access-om3/pull/250#issuecomment-2526544913)

anton-seaice commented on share/CMakeLists.txt:
```diff
@@ -25,6 +24,7 @@ target_sources(OM3_share PRIVATE
   CESM_share/include/shr_assert.h
   CESM_share/src/shr_abort_mod.F90
   CESM_share/src/shr_cal_mod.F90
+  CESM_share/src/shr_const_mod.F90
```
Thanks !

anton-seaice commented:
Build with todays review changes (redeploy 16):

https://github.com/ACCESS-NRI/ACCESS-OM3/pull/33#issuecomment-2611418907

and the test with MOM6-CICE6:

https://github.com/ACCESS-NRI/access-om3-configs/pull/154#issuecomment-2611579022

I also ran an adhoc test with MOM6-CICE6-WW3 which worked with removal of `DEFAULT_2018_ANSWERS = False` from MOM_input

anton-seaice commented:
Ill double check on Tuesday and then merge :-)
